bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/handler/RemoteopenhabThingHandler.java

```diff
@@ -0,0 +1,535 @@
+                    break;
+                case CoreItemFactory.DIMMER:
+                    if (checkStateType(itemName, stateType, "Percent")) {
+                        channelState = new PercentType(state);
+                    }
+                    break;
+                case CoreItemFactory.COLOR:
+                    if (checkStateType(itemName, stateType, "HSB")) {
+                        channelState = HSBType.valueOf(state);
+                    }
+                    break;
+                case CoreItemFactory.DATETIME:
+                    if (checkStateType(itemName, stateType, "DateTime")) {
+                        try {
+                            channelState = new DateTimeType(ZonedDateTime.parse(state, FORMATTER_DATE));
+                        } catch (DateTimeParseException e) {
+                            logger.debug("Failed to parse date {} for item {}", state, itemName);
+                            channelState = null;
+                        }
+                    }
+                    break;
+                case CoreItemFactory.LOCATION:
+                    if (checkStateType(itemName, stateType, "Point")) {
+                        channelState = new PointType(state);
+                    }
+                    break;
+                case CoreItemFactory.IMAGE:
+                    if (checkStateType(itemName, stateType, "Raw")) {
+                        channelState = RawType.valueOf(state);
+                    }
+                    break;
+                case CoreItemFactory.PLAYER:
+                    if (checkStateType(itemName, stateType, "PlayPause")) {
+                        switch (state) {
+                            case "PLAY":
+                                channelState = PlayPauseType.PLAY;
+                                break;
+                            case "PAUSE":
+                                channelState = PlayPauseType.PAUSE;
+                                break;
+                            default:
+                                logger.debug("Unexpected value {} for item {}", state, itemName);
+                                break;
+                        }
+                    }
+                    break;
+                case CoreItemFactory.ROLLERSHUTTER:
+                    if (checkStateType(itemName, stateType, "Percent")) {
+                        channelState = new PercentType(state);
+                    }
+                    break;
+                default:
+                    logger.debug("Item type {} is not yet supported", acceptedItemType);
+                    break;
+            }
+        }
+        if (channelState != null) {
+            updateState(channel.getUID(), channelState);
+            logger.debug("updateState {} with {}", channel.getUID(),
+                    channelState.toFullString().length() < 50 ? channelState.toFullString()
+                            : channelState.toFullString().substring(0, 50) + "...");
```

Comment:
Please cache `channelState.toFullString()` to a local variable.

Reply:
Done